### PR TITLE
Portfolio: validate contact form

### PR DESCRIPTION
**Pull request by Tessyrich**
validated the form to accept only lowercase and also worked with my partner in making some changes.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -9,3 +10,247 @@ document.querySelectorAll('.head-link').forEach((n) => n.addEventListener('click
...
+  }
+  return 'please insert your name';
+};
+
+formVal.addEventListener('submit', (e) => {
+  e.preventDefault();
+  oluchiValidate(inp1.value);
+  oluchiValidate(inp2.value);
+  oluchiValidate(inp3.value);
+});
```
Comment (mckent05) [inline script.js:244]: - In order to follow JS best practices and to keep your code clean and maintainable, I'd suggest you have functions that are needed in your file. What does the method `oluchiValidate` do? 😕 I understand that it checks if the values of the names are empty. But it is currently not working on your site.
- Also, it is very important that your functions or methods are properly named, this helps to make your code more readable. If at all you must use it, I'd suggest you give a better naming to `oluchiValidate`.

File: script.js [hunk 2]
```diff
@@ -9,3 +10,247 @@ document.querySelectorAll('.head-link').forEach((n) => n.addEventListener('click
...
+formVal.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const regex = /[A-Z]/;
+
+  if (!regex.test(emailEl.value)) {
+    formVal.submit();
+  } else {
+    errMsgEmail.innerText = 'Your email should not be in Uppercase';
+  }
+});
```
Comment (mckent05) [inline script.js:256]: - According to the project requirements, the error message should be displayed close to the submit buttons.This is not so in your project. Please try to fix this 👍 👍 

Your project:
formVAlid

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Tessyrich,

# STATUS: CHANGES REQUIRED ♻️ ♻️  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ 
- Your PR title and summary are descriptive 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 3]
```diff
@@ -278,18 +279,20 @@ <h6 class="description">Hello I’m a software developer! I can help you build a
     </section>
   </div>
   <footer class="form-cont" id="form-cont">
-  <form action="https://formspree.io/f/mrgdkyla" method="post">
+  <form action="https://formspree.io/f/mrgdkyla" method="post"  class="form">
```
Comment (topeogunleye) [inline index.html:282]: - [ ] As suggested by the previous reviewer, I kindly suggest that you place your error message close to the submit button.

image

File: script.js [hunk 4]
```diff
@@ -9,3 +10,247 @@ document.querySelectorAll('.head-link').forEach((n) => n.addEventListener('click
...
+const oluchiValidate = (inputbox) => {
+  if (inputbox !== '') {
+    return inputbox;
+  }
+  return 'please insert your name';
+};
+
+formVal.addEventListener('submit', (e) => {
+  e.preventDefault();
+  oluchiValidate(inp1.value);
```
Comment (topeogunleye) [inline script.js:241]: - [ ] As reqquired by the previous reviewer, I kindly suggest that you use a better name for your function here name `oluchiValidate`

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Tessyrich, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors. :heavy_check_mark:
* Followed Github flow :heavy_check_mark:
* Professional README file. :heavy_check_mark:
* PR has a good summary description :heavy_check_mark:

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @Tessyrich :wave: ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request :heavy_multiplication_x: 
Invalid


**_You have submitted a project, that has no linters set up_**.
- [ ] Kindly fix that and resubmit your project for a new review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: index.html (hunk 3 above)
Comment (mckent05) [reply index.html:282]: - Hello @Tessyrich, your email validation is working fine, but the project requirements say that the error message should appear close to the submit button. I'd suggest you put the error message in the same `div` with the submit button and use the `flexbox` attribute to align them however you might please. Please try to fix this 👍 👍 

Example from your code:
 `<div class='btn-cont'><button type="submit" class="get-in see">Get in touch</button><small></small></div>`

CSS: `.btn-cont: {display: "flex";}`

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Tessyrich ,

# STATUS: CHANGES REQUIRED ♻️ ♻️  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Good job implementing some previously suggested changes 👏 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tessyrich** [comment]
It was on top before and I placed it just below it. it never said it should
put it inside a div and it was not specified how close it should be. This
closeness is supposed to be ok. This is my last chance for a review. the
last one failed due to some conflict with my main branch. Will this make me
repeat a week because my chances have been exhausted?

**Review by bimbolabuari — COMMENTED** [review]
Hello @Tessyrich 👋,

_This is an additional review as a result of the second opinion request._

- As required, the error message for the form validation shows up close to the submit button. 👍 

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Pull request merged**